plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskList.php

```diff
@@ -171,7 +171,7 @@ public function is_hidden() {
 	 * @return bool
 	 */
 	public function is_visible() {
-		if ( ! $this->visible ) {
+		if ( ! $this->visible || ! count( $this->get_viewable_tasks() ) > 0 ) {
```

Comment:
Adding having viewable tasks as part of the condition for `tasklist->is_visible` to resolve a bug with tracks events being fired even when a tasklist was not visible.